### PR TITLE
Added a CMakeLists file

### PR DESCRIPTION
**Pull request by oneechanhax**
this makes adding this project to other cmake projects easier. It could even replace the simple build system you currently have.

### CODE REVIEW

**Comment by oneechanhax** [comment]
Not sure if the other requirements were optional but i can easily add those as they are supported by cmake

**Comment by yourWaifu** [comment]
This is trying to solve #13, correct? Also as for librarys, there are no requirements, you can compile and use my library without any other librarys and use it. Anyway, you weren't the first to do pull request on this idea, but can you please make sure that tick all the points from issue #13.

**Comment by oneechanhax** [comment]
It wasnt my initial intention to make cross platform for windows and such, but cmake already handles that for you, and as for the selection of openssl and such, you could always just compile without them and link them with your own lib, for example i compile sleepy discord without libs linked, then in my executable i link in openssl to get that functionality.
I dont know if thats how this works thoough. 
But cmake already handles visual studio project files as well as unix make files so that point is down.

**Comment by yourWaifu** [comment]
I'll do some testing later and see if any issues come up before doing a merge as I don't want a repeat of the last attempt.

**Comment by oneechanhax** [comment]
The cmake lists file now properly works. It does exactly what it is supposed to.
You can use it in visual studio on windows. and linux/mac with their compiler systems like GNU/make

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,25 +1,19 @@
...
 
 # Open SSL
 find_package(OpenSSL REQUIRED)
 
+# Not reccomended but I do it anyway
+file(GLOB_RECURSE SOURCE_FILES "${CMAKE_SOURCE_DIR}/sleepy_discord/*.c*")
 # why static >_>
 add_library(sleepydiscord STATIC "${SOURCE_FILES}")
 
+target_include_directories(sleepydiscord PUBLIC "include/sleepy_discord/" "include/sleepy_discord/IncludeNonexistent/")
```
Comment (yourWaifu) [inline CMakeLists.txt]: I think the include directories should be
```
include
deps
deps/include
include/sleepy_discord/IncludeNonexistent```

**Comment by yourWaifu** [comment]
I'll try to get this tested and if it passes, I'll do a merge

File: CMakeLists.txt (hunk 1 above)
Comment (oneechanhax) [reply CMakeLists.txt]: shouldnt deps and deps/include be done by the user when using an already compiled sleepy-discord library?
Also, i dont see the folders for deps anywhere.

**Comment by oneechanhax** [comment]
Here is an example cmake file to use with projects using sleepy-discord.
You dont even need a compiled library as it compiles it for you.
https://github.com/oneechanhax/nekobot/blob/master/CMakeLists.txt

File: CMakeLists.txt (hunk 1 above)
Comment (yourWaifu) [reply CMakeLists.txt]: well it's created by the user or the setup script. It's used for a place to have all the library the user will use with sleepy discord so maybe it shouldn't be there.

**Pull request merged**